opentelemetry-api/src/opentelemetry/propagation/__init__.py

```diff
@@ -0,0 +1,144 @@
+# Copyright 2019, OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import typing
+
+import opentelemetry.context.propagation.httptextformat as httptextformat
+import opentelemetry.trace as trace
+from opentelemetry.context import Context
+from opentelemetry.context.propagation import (
+    ContextT,
+    DefaultHTTPExtractor,
+    DefaultHTTPInjector,
+    Getter,
+    Setter,
+)
+
+
+def extract(
+    carrier: ContextT,
+    context: typing.Optional[Context] = None,
+    extractors: typing.Optional[
+        typing.List[httptextformat.HTTPExtractor]
+    ] = None,
+    get_from_carrier: typing.Optional[Getter[ContextT]] = None,
+) -> typing.Optional[Context]:
+    """Load the parent SpanContext from values in the carrier.
+
+    Using the specified HTTPExtractor, the propagator will
+    extract a SpanContext from the carrier. If one is found,
+    it will be set as the parent context of the current span.
+
+    Args:
+        get_from_carrier: a function that can retrieve zero
+            or more values from the carrier. In the case that
+            the value does not exist, return an empty list.
+        carrier: and object which contains values that are
+            used to construct a SpanContext. This object
+            must be paired with an appropriate get_from_carrier
+            which understands how to extract a value from it.
+    """
+    if context is None:
+        context = Context.current()
+    if extractors is None:
+        extractors = get_http_extractors()
+
+    for extractor in extractors:
+        # TODO: improve this
```

Comment:
add try/except. return on the first successful extraction

Reply:
What about if there are contexts for multiple concerns on the same carrier?
I think this should loop through all the extractors logging an error/warning one fails.